cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ModelUtils.java

```diff
@@ -812,4 +813,62 @@ public static List<String> generateAllServiceDnsNames(String namespace, String s
 
         return dnsNames;
     }
+
+    /**
+     * Validate cpu and memory resources.
+     * Early resources validation avoids triggering any pod operation with invalid configuration.
+     * 
+     * For both cpu and memory, this method checks that request is greater than zero, 
+     * limit is greater than zero and limit is greater than or equal to the request.
+     * 
+     * @param resources Resources configuration.
+     * @param owner Resources owner.
+     */
+    public static void validateComputeResources(ResourceRequirements resources, String owner) {
+        List<String> errors = ModelUtils.validateComputeResources(resources, "cpu", owner);
+        errors.addAll(ModelUtils.validateComputeResources(resources, "memory", owner));
+        if (errors.size() > 0) {
+            throw new InvalidResourceException(errors.toString());
+        }
+    }
+
+    /**
+     * Validate compute resources for a given owner.
+     * 
+     * This method checks that request is greater than zero, limit is greater than zero 
+     * and limit is greater than or equal to the request.
+     * 
+     * @param resources Compute resources configuration.
+     * @param type Compute resources type.
+     * @param owner Compute resources owner.
+     * 
+     * @return Error set.
+     */
+    private static List<String> validateComputeResources(ResourceRequirements resources, String type, String owner) {
+        List<String> errors = new ArrayList<>();
+        String oid = owner != null && !owner.isBlank() ? owner : "component";
+        if (resources != null) {
+            if (resources.getRequests() != null && resources.getRequests().get(type) != null) {
+                Quantity request = resources.getRequests().get(type);
+                if (request != null && request.getNumericalAmount().compareTo(BigDecimal.ZERO) <= 0) {
+                    errors.add(String.format("%s %s request must be > zero", oid, type));
+                }
+            }
+            if (resources.getLimits() != null && resources.getLimits().get(type) != null) {
+                Quantity limit = resources.getLimits().get(type);
+                if (limit.getNumericalAmount().compareTo(BigDecimal.ZERO) <= 0) {
+                    errors.add(String.format("%s %s limit must be > zero", oid, type));
+                }
+            }
+            if (resources.getRequests() != null && resources.getRequests().get(type) != null
+                    && resources.getLimits() != null && resources.getLimits().get(type) != null) {
+                Quantity limit = resources.getLimits().get(type);
+                Quantity request = resources.getRequests().get(type);
+                if (request != null && request.getNumericalAmount().compareTo(limit.getNumericalAmount()) > 0) {
+                    errors.add(String.format("%s %s request must be <= limit", oid, type));
+                }
+            }
+        }
+        return errors;
+    }
```

Comment:
Maybe you can consider better spacing for improved readablity? I would for example space it like this:

```suggestion
    private static List<String> validateComputeResources(ResourceRequirements resources, String type, String owner) {
        List<String> errors = new ArrayList<>();
        String oid = owner != null && !owner.isBlank() ? owner : "component";
        
        if (resources != null) {
            if (resources.getRequests() != null && resources.getRequests().get(type) != null) {
                Quantity request = resources.getRequests().get(type);
                
                if (request != null && request.getNumericalAmount().compareTo(BigDecimal.ZERO) <= 0) {
                    errors.add(String.format("%s %s request must be > zero", oid, type));
                }
            }
            
            if (resources.getLimits() != null && resources.getLimits().get(type) != null) {
                Quantity limit = resources.getLimits().get(type);
                
                if (limit.getNumericalAmount().compareTo(BigDecimal.ZERO) <= 0) {
                    errors.add(String.format("%s %s limit must be > zero", oid, type));
                }
            }
            
            if (resources.getRequests() != null && resources.getRequests().get(type) != null
                    && resources.getLimits() != null && resources.getLimits().get(type) != null) {
                Quantity limit = resources.getLimits().get(type);
                Quantity request = resources.getRequests().get(type);
                
                if (request != null && request.getNumericalAmount().compareTo(limit.getNumericalAmount()) > 0) {
                    errors.add(String.format("%s %s request must be <= limit", oid, type));
                }
            }
        }
        
        return errors;
    }
```

Just a suggestion whcih might improve the code look.